File: C/projects/DALEC_CODE/DALEC_ALL/HYDROLOGY_MODULES/CONVERTERS/HYDROFUN_MOI2PSI.c
#pragma once

/*
Convert volumetric soil moisture (m3/m3) to potential (MPa)
Inputs:
sm - Soil moisture (m3/m3)
psi_porosity - porosity potential (MPa)
b - retention parameter
*/

double HYDROFUN_MOI2PSI(double sm, double psi_porosity, double b){

double psi=psi_porosity*(pow((1/sm),b));
 
return psi;
}

File: PYTHON/projects/CARDAMOM_MODELS/DALEC/DALEC_ALL/HYDROLOGY_MODULES/CONVERTERS/HYDROFUN_MOI2PSI.py
import jax.numpy as jnp

def HYDROFUN_MOI2PSI(moi, psi_porosity, b):
    """Convert volumetric soil moisture (m3/m3) to potential (MPa)"""
    return psi_porosity * jnp.power(1.0 / jnp.maximum(moi, 1e-9), b)
